### DISCUSSION TITLE
کاهش حجم دیتابیس در Mysql

### DISCUSSION THREAD

**Discussion by amirgi73** [Gozargah/Marzban#396] [Show and tell]
اگر به Mysql مهاجرت کرده باشید متوجه مصرف بالای RAM و همچنین حجم بالای دیتابیس نسبت به Sqlite خواهید شد. بحث حجم مخصوصا برای گرفتن بک‌آپ با ورژن فعلی [AC-Backup](https://github.com/AC-Lover/backup) شما رو دچار مشکل خواهد کرد. چرا که API تلگرام اجاره آپلود فایل با حجم بیشتر از 50 مگابایت رو نمیده.

به صورت پیش‌فرض در Mysql قابلیت performance-schema فعاله. این قابلیت برای دیباگ کردن و اطلاعات آماری دیتابیس خیلی مفیده. ولی در استفاده فعلی ما به نظر من نیازی بهش نیست و با غیرفعال کردنش میشه تا حد زیادی در مصرف RAM صرفه جویی کرد. به علاوه اینکه حجم بک‌آپ‌های دیتابیستون از چند صد مگابایت به حدود 10 مگابایت کاهش پیدا خواهد کرد. 

<details><summary>نکته</summary>
<p>

اگر از اسکریپت AC-Backup استفاده می کنید. قبل از شروع اون رو به آخرین نسخه آپدیت کنید.

</p>
</details> 

### مرحله اول
در فایل `docker-compose.yml` در قسمت مربوط به `mysql` این درایو رو اضافه کنید.

```dockerfile
- /var/lib/marzban/mysql-config:/etc/mysql/conf.d
``` 
به عنوان مثال فایلتون این شکلی می‌شه:

```dokcerfile
services:
  mysql:
    image: mysql:latest
    restart: always
    env_file: .env
    network_mode: host
    environment:
      MYSQL_DATABASE: marzban
    volumes:
      - /var/lib/marzban/mysql:/var/lib/mysql
      - /var/lib/marzban/mysql-config:/etc/mysql/conf.d

``` 

### مرحله دوم
یک پوشه جدید به اسم `mysql-config` در مسیر `var/lib/marzban/` ایجاد کنید.

```shell
mkdir /var/lib/marzban/mysql-config
``` 

در پوشه‌ای که تازه ایجاد کردید فایلی به اسم `my.cnf` با محتویات زیر بسازید.

```
[mysqld]
performance_schema = 0
``` 

<details><summary>نکته</summary>

برای ایجاد فایل بالا می‌توانید از این دستور استفاده کنید.

```shell
echo -e "[mysqld]\nperformance_schema = 0" > /var/lib/marzban/mysql-config/my.cnf
```
</details> 

### مرحله سوم
مرزبان رو ریستارت کنید.
```shell
marzban down
marzban up
```

**Comment by DrAmirB** [comment]
با استفاده از این روش اون برتری که mysql نسبت به sqlite داره بی اثر نمیشه؟

**Reply by Ashaxer** [reply]
به غیر از قابل تنظیم بودن دیتابیس، برتری دیگه ای هم داره؟

**Reply by MirzaT1992** [reply]
اره اگر روی SQLite یوزر بالا بزنی اذیت میکنه و خیلی چیز های دیگه

**Reply by MirzaT1992** [reply]
مثلا کوئری
